src/screens/transaction-details/components/TransactionDetailsStatusActionsAndTimestampSection.tsx

```diff
@@ -36,36 +35,37 @@ export const TransactionDetailsStatusActionsAndTimestampSection: React.FC<Props>
   const canBeCancelled = canBeResubmitted && status !== TransactionStatusTypes.cancelling;
 
   const menuConfig = useMemo(
-    () => ({
-      menuTitle: '',
-      menuItems: [
-        ...(canBeResubmitted
-          ? [
-              {
-                actionKey: 'speedUp',
-                actionTitle: i18n.t(i18n.l.transaction_details.actions_menu.speed_up),
-                icon: {
-                  iconType: 'SYSTEM',
-                  iconValue: 'speedometer',
+    () =>
+      ({
+        menuTitle: '',
+        menuItems: [
+          ...(canBeResubmitted
+            ? [
+                {
+                  actionKey: 'speedUp',
+                  actionTitle: i18n.t(i18n.l.transaction_details.actions_menu.speed_up),
+                  icon: {
+                    iconType: 'SYSTEM',
+                    iconValue: 'speedometer',
+                  },
                 },
-              },
-            ]
-          : []),
-        ...(canBeCancelled
-          ? [
-              {
-                actionKey: 'cancel',
-                actionTitle: i18n.t(i18n.l.transaction_details.actions_menu.cancel),
-                menuAttributes: ['destructive'],
-                icon: {
-                  iconType: 'SYSTEM',
-                  iconValue: 'xmark.circle',
+              ]
+            : []),
+          ...(canBeCancelled
+            ? [
+                {
+                  actionKey: 'cancel',
+                  actionTitle: i18n.t(i18n.l.transaction_details.actions_menu.cancel),
+                  menuAttributes: ['destructive' as const],
```

Comment:
is this necessary to cast as `const`

Reply:
yes, it considers 'destructive' as `string` but the type wants the it as the string literal type `destructive`, same stuff for the others `as const` or `satisfies` added, I expanded the type a lil bit, but in these cases opted to keep it like this so we have better auto complete